### PR TITLE
SONAR-6397 Execute blame only when needed

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/scm/ScmSensor.java [hunk 1]
```diff
@@ -95,11 +93,12 @@ public void execute(SensorContext context) {
...
     for (InputFile f : fs.inputFiles(fs.predicates().all())) {
-      if (configuration.forceReloadAll()) {
+      if (configuration.forceReloadAll() || f.status() != Status.SAME) {
         addIfNotEmpty(filesToBlame, f);
       } else {
-        FileData fileData = projectSettings.fileData(projectDefinition.getKeyWithBranch(), f.relativePath());
-        if (f.status() != Status.SAME || fileData == null || fileData.revision() != null) {
+        // File status is SAME so that mean fileData exists
+        FileData fileData = projectRepositories.fileData(projectDefinition.getKeyWithBranch(), f.relativePath());
+        if (StringUtils.isEmpty(fileData.revision())) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/scm/ScmSensor.java]: BLOCKER NullPointerException might be thrown as 'fileData' is nullable here [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

**Comment by dbmeneses** [comment]
LGTM - there is the possible NPE reported

**Comment by henryju** [comment]
@dbmeneses  NPE is a won't fix (look at the comment)

**Pull request merged**